programs/rewards/src/utils.rs

```diff
@@ -238,6 +241,27 @@ impl AccountLoader {
         Err(ProgramError::MissingRequiredSignature)
     }
 
+    /// Checks if next account is a signer AND is derived from the Staking program
+    pub fn next_signer_from_staking_program<'a, 'b, I: Iterator<Item = &'a AccountInfo<'b>>>(
+        iter: &mut Enumerate<I>,
+    ) -> Result<I::Item, ProgramError> {
+        #[cfg(not(feature = "testing"))]
+        {
+            let (idx, acc) = iter.next().ok_or(ProgramError::NotEnoughAccountKeys)?;
+            if acc.is_signer && acc.key.eq(&STAKING_PROGRAM_REGISTRAR) {
+                return Ok(acc);
+            }
+            msg!("Account #{}:{} missing signature", idx, acc.key,);
```

Comment:
if the signer is not staking program the message is a bit misleading, I'd change it or separate the checks for signer and key equality